subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java

```diff
@@ -178,12 +179,17 @@ public EntitlementState getState() {
         }
 
         final SubscriptionBaseTransition pendingTransition = getPendingTransition();
-        if (pendingTransition != null) {
-            return EntitlementState.PENDING;
+        if (pendingTransition != null && pendingTransition.getEffectiveTransitionTime().isAfter(clock.getUTCNow())) {
```

Comment:
I would expect that if `pendingTransition != null` then, it is always in the future, because of how it was computed, i.e iterator looking through `transitions` with [FUTURE_ONLY](https://github.com/killbill/killbill/blob/master/subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java#L317)